src/lib/kernel/LinuxProcess.cc

```diff
@@ -22,18 +23,17 @@ LinuxProcess::LinuxProcess(const std::vector<std::string>& commandLine,
       commandLine_(commandLine) {
   // Parse ELF file
   assert(commandLine.size() > 0);
-  Elf elf(commandLine[0]);
+
+  Elf elf(commandLine[0], &processImage_);
   if (!elf.isValid()) {
     return;
   }
   isValid_ = true;
 
   entryPoint_ = elf.getEntryPoint();
 
-  span<char> elfProcessImage = elf.getProcessImage();
-
   // Align heap start to a 32-byte boundary
-  heapStart_ = alignToBoundary(elfProcessImage.size(), 32);
+  heapStart_ = alignToBoundary(elf.getProcessImage().size(), 32);
```

Comment:
Could change this to a processImageSize_ getter instead

Reply:
PTAL